sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -47,6 +52,12 @@ object Klaviyo {
                 .applicationContext(applicationContext)
                 .build()
         )
+
+        val application = applicationContext.applicationContext as? Application
```

Comment:
Here is the [documentation](https://developer.android.com/reference/android/content/Context.html#getApplicationContext()) on `applicationContext`

In earlier speccing last year, I shied away from using this method to access the `Application` object because of a stackoverflow post from 2011 that warned off using it. Now that I've read more about it, I think that post was incorrect. For example, this description from official docs describes exactly what we're doing here: 

> Return the context of the single, global Application object of the current process. This generally should only be used if you need a Context whose lifecycle is separate from the current context, that is tied to the lifetime of the process rather than the current component.